index.ts

```diff
@@ -28,24 +28,68 @@ function printSbtTask(task: string, cwd?: string): Promise<string> {
       if (code !== 0)
         reject(new Error(`sbt invocation for Scala.js compilation failed with exit code ${code}.`));
       else
-        resolve(fullOutput.trimEnd().split('\n').at(-1)!);
+        resolve(fullOutput.trimEnd().split('\n').slice(-tasks.length));
     });
   });
 }
 
+export interface Subproject {
+  projectID: string | null,
+  uriPrefix: string,
+}
+
 export interface ScalaJSPluginOptions {
   cwd?: string,
   projectID?: string,
   uriPrefix?: string,
+  subprojects?: Array<Subproject>,
 }
 
-export default function scalaJSPlugin(options: ScalaJSPluginOptions = {}): VitePlugin {
-  const { cwd, projectID, uriPrefix } = options;
+function extractSubprojects(options: ScalaJSPluginOptions): Array<Subproject> {
+  if (options.subprojects) {
+    if (options.projectID || options.uriPrefix) {
+      throw new Error("If you specify subprojects, you cannot specify projectID / uriPrefix")
+    }
+    return options.subprojects;
+  } else {
+    return [
+      {
+        projectID: options.projectID || null,
+        uriPrefix: options.uriPrefix || 'scalajs',
+      }
+    ];
+  }
+}
+
+function mapBy<T, K>(a: Array<T>, f: ((item: T) => K), itemName: string): Map<K, T> {
+  const out = new Map<K, T>();
+  a.forEach((item) => {
+    const key: K = f(item);
+    if (out.has(key)) {
+      throw Error("Duplicate " + itemName + " " + key + ".");
+    } else {
+      out.set(key, item);
+    }
+  });
+  return out;
+}
+
+function zip<T, U>(a: Array<T>, b: Array<U>): Array<[T, U]> {
+  if (a.length != b.length) {
+    throw new Error("length mismatch: " + a.length + " ~= " + b.length)
+  }
+  return a.map((item, i) => [item, b[i]]);
+}
 
-  const fullURIPrefix = uriPrefix ? (uriPrefix + ':') : 'scalajs:';
+export default function scalaJSPlugin(options: ScalaJSPluginOptions = {}): VitePlugin {
+  const { cwd } = options;
+  const subprojects = extractSubprojects(options);
+  // This also checks for duplicates
+  const spByProjectID = mapBy(subprojects, (p) => p.projectID, "projectID")
+  const spByUriPrefix = mapBy(subprojects, (p) => p.uriPrefix, "uriPrefix")
```

Comment:
These variables are unused. So instead of a `mapBy`, consider simplifying to *only* check duplicates in an array:
```typescript
function checkNoDuplicates<T>(a: Array<T>, itemName: string): void
```
then call it as
```suggestion
  checkNoDuplicates(subprojects.map((p) => p.projectID), "projectID");
  checkNoDuplicates(subprojects.map((p) => p.uriPrefix), "uriPrefix");
```

Reply:
I am a bit undecided there. Sure, this can be the way. OTOH, I suspect those variables might be needed soon.